plugins/orchestrator/src/components/ExecuteWorkflowPage/StepperForm.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+
+import { Content, StructuredMetadataTable } from '@backstage/core-components';
+import { JsonValue } from '@backstage/types';
+
+import {
+  Box,
+  Button,
+  Paper,
+  Step,
+  StepContent,
+  StepLabel,
+  Stepper,
+  Typography,
+} from '@material-ui/core';
+import { withTheme } from '@rjsf/core-v5';
+import { Theme as MuiTheme } from '@rjsf/material-ui-v5';
+import validator from '@rjsf/validator-ajv8';
+import { JSONSchema7 } from 'json-schema';
+
+import SubmitButton from '../SubmitButton/SubmitButton';
+
+const MuiForm = withTheme<Record<string, JsonValue>>(MuiTheme);
+
+const ReviewStep = ({
+  busy,
+  formDataObjects,
+  handleBack,
+  handleReset,
+  handleExecute,
+}: {
+  busy: boolean;
+  formDataObjects: Record<string, JsonValue>[];
+  handleBack: () => void;
+  handleReset: () => void;
+  handleExecute: () => void;
+}) => {
+  const combinedFormData = React.useMemo(
+    () =>
+      formDataObjects.reduce<Record<string, JsonValue>>(
+        (prev, cur) => ({ ...prev, ...cur }),
+        {},
+      ),
+    [formDataObjects],
+  );
+  return (
+    <Content>
+      <Paper square elevation={0}>
+        <Typography variant="h6">Review and run</Typography>
+        <StructuredMetadataTable dense metadata={combinedFormData} />
+        <Box mb={4} />
+        <Button onClick={handleBack} disabled={busy}>
+          Back
+        </Button>
+        <Button onClick={handleReset} disabled={busy}>
+          Reset
+        </Button>
+        <SubmitButton handleClick={handleExecute} submitting={busy}>
+          Run
+        </SubmitButton>
+      </Paper>
+    </Content>
+  );
+};
+
+const StepperForm = ({
+  refSchemas,
+  handleExecute,
+  isExecuting,
+}: {
+  refSchemas: JSONSchema7[];
+  handleExecute: (
+    getParameters: () => Record<string, JsonValue>,
+  ) => Promise<void>;
+  isExecuting: boolean;
+}) => {
+  const [activeStep, setActiveStep] = React.useState(0);
+  const handleBack = () => setActiveStep(activeStep - 1);
+
+  const [formDataObjects, setFormDataObjects] = React.useState<
+    Record<string, JsonValue>[]
+  >([]);
+
+  const getFormData = () =>
+    formDataObjects.reduce<Record<string, JsonValue>>(
+      (prev, curFormObject) => ({ ...prev, ...curFormObject }),
+      {},
+    );
+
+  const resetFormDataObjects = React.useCallback(
+    () =>
+      setFormDataObjects(
+        refSchemas.reduce<Record<string, JsonValue>[]>(
+          prev => [...prev, {}],
+          [],
+        ),
+      ),
+    [refSchemas],
+  );
+
+  React.useEffect(() => {
+    resetFormDataObjects();
+  }, [resetFormDataObjects]);
+
+  return (
+    <>
+      <Stepper activeStep={activeStep} orientation="vertical">
+        {refSchemas.map((schema, index) => (
+          <Step key={index}>
```

Comment:
What should I use instead? The property schema.$id is not guaranteed
Since this list won't be reordered and items won't be deleted and added, I don't see a reason to not use the index

Reply:
What about `schema.$id ?? index` ? At least, the sonarcloud issue is gone this way. 😃